### PR TITLE
Jvenstad/old instance data cleanup

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

**Comment by jonmv** [comment]
Makes it clunky to completely delete applications with more than one instance, but there aren't any of those yet. Will fix this this in a more elegant way shortly.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 1]
```diff
@@ -146,8 +146,16 @@
         Once.after(Duration.ofMinutes(1), () -> {
             Instant start = clock.instant();
             int count = 0;
-            for (Application application : curator.readApplications()) {
-                lockApplicationIfPresent(application.id(), this::store);
+            for (TenantAndApplicationId id: curator.readApplicationIds()) {
+                lockApplicationIfPresent(id, application -> {
+                    if (id.tenant().value().startsWith("by-"))
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: In the future we should disallow creation of by tenants altogether and remove this.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 2]
```diff
@@ -718,6 +718,7 @@ public void deleteApplication(TenantAndApplicationId id, Optional<Credentials> c
         if (instances.size() > 1)
             throw new IllegalArgumentException("Could not delete application; more than one instance present: " + instances);
 
+        lockApplicationOrThrow(id, application -> store(application.with(DeploymentSpec.empty)));
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Hopefully not a long-lived solution. 🙂

**Review by mpolden — APPROVED** [review]

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Hehe.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: No, this all needs to be solved properly. Deletion at application level, with a confirmatino dialogue with a list of all instances to be deleted, or something.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Absolutely.

**Pull request merged**